learning/prompts/documentation-lookup/31_advanced_batched_dofn.md

```diff
@@ -0,0 +1,35 @@
+Prompt:
+What is a Batched DoFn in Apache Beam?
+Response:
+In Apache Beam, a [Batched DoFn](https://beam.apache.org/documentation/programming-guide/#batched-dofns) is a [DoFn](https://beam.apache.org/documentation/programming-guide/#pardo) that can operate on batches of logical elements. These `DoFns` leverage popular Python vectorized libraries like [NumPy](https://numpy.org/), [SciPy](https://scipy.org/) and [pandas](https://pandas.pydata.org/) to efficiently process batches of elements in a `PCollection` rather than processing each element individually.
+
+Batched `DoFns` are only available in the Apache Beam Python SDK.
+
+You can provide element-wise, batched or both implementation for a `DoFn`
+
```

Comment:
```suggestion
For some ‘DoFn’s, you can provide both a batched and an element-wise implementation of your desired logic by defining both ‘process’ and ‘process_batch’ methods. 

Here is a simple example of a flexible ‘DoFn’ with both batched and element-wise implementations:

```

Reply:
In the documentation, they do not say it is possible for all DoFns, they say "for some"